timemachine/fe/rbfe.py

```diff
@@ -122,205 +129,150 @@ def setup_initial_states(st, host_config, temperature, lambda_schedule, seed, mi
-            group_idxs = get_group_indices(get_bond_list(system.bond))
-            baro = MonteCarloBarostat(len(hmr_masses), 1.0, temperature, group_idxs, 15, run_seed + 1)
-            x0 = np.concatenate([host_conf, ligand_conf])
-            box0 = host_config.box
-        else:
-            # run a vacuum simulation
-            system = st.setup_intermediate_state(lamb)
-            combined_masses = np.array(st.combine_masses())
-            potentials = system.get_U_fns()
-            hmr_masses = model_utils.apply_hmr(combined_masses, system.bond.get_idxs())
-            baro = None
-            x0 = ligand_conf
-            box0 = np.eye(3, dtype=np.float64) * 10  # make a large 10x10x10nm box
-
-        # minimize water box around the ligand by 4D-decoupling
-
-        # hmr masses should be identical throughout the lambda schedule
-        # bond idxs should be the same at the two end-states, note that a possible corner
-        # case with bond breaking may seem to be problematic:
-
-        # 0 1 2    0 1 2
-        # C-O-C -> C.H-C
-
-        # but this isn't an issue, since hydrogens will only ever be terminal atoms
-        # and core hydrogens that are mapped to heavy atoms will take the mass of the
-        # heavy atom (thereby not triggering the mass repartitioning to begin with).
+            x0, box0, hmr_masses, potentials, baro = setup_in_env(
+                st, host, host_config, ligand_conf, lamb, temperature, init_seed
+            )
 
-        # but its reasonable to be skeptical, so we also assert consistency through the lambda
-        # schedule as an extra sanity check.
+        # provide a different run_seed for every lambda window,
+        # but in a way that should be symmetric for
+        # A -> B vs. B -> A edge definitions
+        run_seed = int(seed + hash(bytes().join([np.array(p.params).tobytes() for p in potentials]))) % 10000
+        # the constant is arbitrary, but see
+        # https://github.com/proteneer/timemachine/commit/e1f7328f01f427534d8744aab6027338e116ad09
 
+        # initialize velocities
         v0 = np.zeros_like(x0)  # tbd resample from Maxwell-boltzman?
+
+        # determine ligand idxs
         num_ligand_atoms = len(ligand_conf)
         num_total_atoms = len(x0)
         ligand_idxs = np.arange(num_total_atoms - num_ligand_atoms, num_total_atoms)
 
+        # initialize Langevin integrator
         dt = 2.5e-3
         friction = 1.0
         intg = LangevinIntegrator(temperature, dt, friction, hmr_masses, run_seed)
 
+        # pack into state
         state = InitialState(potentials, intg, baro, x0, v0, box0, lamb, ligand_idxs)
         initial_states.append(state)
 
-    all_coords = optimize_coordinates(initial_states, min_cutoff=min_cutoff)
-    for state, coords in zip(initial_states, all_coords):
-        state.x0 = coords
+    # minimize environment atoms within min_cutoff of the ligand
```

Comment:
```suggestion
    # minimize ligand and environment atoms within min_cutoff
```

A bit ambiguous that `optimize_coordinates` is optimizing the environment and the ligand

Reply:
Applied in https://github.com/proteneer/timemachine/pull/936/commits/c244e4a772b8206c19858101902cd94632273dee